hcl/hcl_import_test.go

```diff
@@ -0,0 +1,457 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hcl_test
+
+import (
+	"testing"
+
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
+)
+
+func TestHCLImport(t *testing.T) {
+	for _, tc := range []testcase{
+		{
+			name: "import with label - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import "something" {
+						source = "bleh"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("cfg.tm", start(1, 8, 7), end(1, 19, 18))),
+				},
+			},
+		},
+		/*{
+			name: "import missing source - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import {
+
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("cfg.tm", start(1, 8, 7), end(1, 8, 18))),
+				},
+			},
+		},*/
```

Comment:
Is this supposed to be commented ? (like a future TODO ?)

Reply:
Ah seems like a duplicate from "import with non-existent file - fails"

Reply:
ah duh..it is missing souce attribute..not the source file itself is missing :laughing: 

![](https://raw.githubusercontent.com/katcipis/memes/master/duh.jpeg)